hw/i2c/smbus_eeprom.c

```diff
@@ -84,6 +85,8 @@ static void eeprom_write_data(SMBusDevice *dev, uint8_t cmd, uint8_t *buf, int l
     len -= n;
     if (len)
         memcpy(eeprom->data, buf + n, len);
+
+    save_eeprom(eeprom->data);
```

Comment:
How would QEMU know *which* eeprom to use?
You'll need some QOM magic here to figure out who this eeprom belongs to.

I can't really help with it either, because of my hatred and resulting inexperience with QOM.